src/simulation/m_start_up.fpp

```diff
@@ -1353,10 +1353,8 @@ contains
         ! Associate pointers for serial or parallel I/O
         if (parallel_io .neqv. .true.) then
             s_read_data_files => s_read_serial_data_files
```

Comment:
can we just wrap all of this logic in a subroutine instead of only the `write` subroutine? (include the `read` one as well)